### PR TITLE
TagsInput widget

### PR DESCRIPTION
**Pull request by martinRenou**
tagsinput

### CODE REVIEW

**Comment by vidartf** [comment]
This looks like an interesting widget, but it isn't obvious to me if this belongs in controls or in a third-party package. Do we have a policy here?

**Comment by martinRenou** [comment]
There is more in https://github.com/martinRenou/tagsinput. I can update this PR if needed

**Comment by martinRenou** [comment]
> This looks like an interesting widget, but it isn't obvious to me if this belongs in controls or in a third-party package. Do we have a policy here?

I don't know myself, and I guess it will be the same question for the TextSlider PR. Maybe @jasongrout @SylvainCorlay you can answer this?

File: ipywidgets/widgets/widget_tagsinput.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    _view_name = Unicode('TagsInputView').tag(sync=True)
+
+    value = List(Unicode(), help='List of string tags').tag(sync=True)
+    tag_style = CaselessStrEnum(
+        values=['primary', 'success', 'info', 'warning', 'danger', ''], default_value='',
+        help="""Use a predefined styling for the tags.""").tag(sync=True)
+
+
+@register
+class ColorsInput(TagsInputBase):
```
Comment (SylvainCorlay) [inline ipywidgets/widgets/widget_tagsinput.py]: ColorTags / TextTags / IntTags / FloatTags?

**Comment by SylvainCorlay** [comment]
Vidar: Neutral for inclusion
Jason: Neutral
Sylvain: Slightly in favor (don't see a better place for it).

Line of divide for inclusion?
------------------
 - browser standard control -> yes
 - "very common UX control" ?

**Comment by Alexboiboi** [comment]
Hi there, 

thanks @martinRenou, I really like the widget and I am looking forward to see it in the main package!
@jasongrout, any chance this is going to be merged soon?

**Comment by SylvainCorlay** [comment]
@martinRenou, could you rebase this one? We decided to go ahead with this PR.

File: packages/controls/src/widget_tagsinput.ts [hunk 2]
```diff
@@ -0,0 +1,844 @@
...
+  /**
+   * Create the Color tag
+   */
+  createTag(value: string, index: number, selected: boolean): HTMLDivElement {
+    const tag = document.createElement('div');
+    const color = value;
+    const darkerColor: string = d3Color
+      .color(value)!
+      .darker()
+      .toString();
```
Comment (SylvainCorlay) [inline packages/controls/src/widget_tagsinput.ts:679]: Ping @martinRenou I had to add `!` because `d3Color.color(value)` may be null.

Thoughts?

Comment (martinRenou) [reply packages/controls/src/widget_tagsinput.ts:679]: Hmm I don't understand that code, does this actually compile?

Comment (SylvainCorlay) [reply packages/controls/src/widget_tagsinput.ts:679]: Yes, it returns `null` if `d3Color.color(value)` is null.

Comment (martinRenou) [reply packages/controls/src/widget_tagsinput.ts:679]: What is `d3Color.color(value)!`? Shouldn't this raise a `SyntaxError: Unexpected token`?

Comment (SylvainCorlay) [reply packages/controls/src/widget_tagsinput.ts:679]: `d3Color.color(value)!.darker()`

Comment (martinRenou) [reply packages/controls/src/widget_tagsinput.ts:679]: Sorry, this does not make sense to me

Comment (martinRenou) [reply packages/controls/src/widget_tagsinput.ts:679]: Just for the record, this is a TypeScript operator: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

**Pull request merged**